### PR TITLE
feat(er/attachment): support new attachment sdk

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
add a new attachment sdk and support a list method.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/er/v3/attachments/requests.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	Statuses []string `q:"state"`
+	// The resource types to be filtered.
+	// + vpc: virtual private cloud
+	// + vpn: vpn gateway
+	// + vgw: virtual gateway of cloud private line
+	// + peering: Peering connection, through the cloud connection (CC) to load enterprise routers in different regions
+	//   to create a peering connection
+	ResourceTypes []string `q:"resource_type"`
+	// The resource IDs to be filtered.
+	resourceIds []string `q:"resource_id"`
```
Comment (deer-hang) [inline openstack/er/v3/attachments/requests.go:29]:
```suggestion
	ResourceIds []string `q:"resource_id"`
```

Comment (Lance52259) [reply openstack/er/v3/attachments/requests.go]: fixed.

**Comment by Lance52259** [comment]
the description of the parameter `next_marker` has been updated, and the function has been validated in the resource test
```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccAttachmentsDataSource_testMarker'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccAttachmentsDataSource_testMarker -timeout 360m -parallel 4
=== RUN   TestAccAttachmentsDataSource_testMarker
=== PAUSE TestAccAttachmentsDataSource_testMarker
=== CONT  TestAccAttachmentsDataSource_testMarker
--- PASS: TestAccAttachmentsDataSource_testMarker (113.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        113.553s
```

**Pull request merged**